app/models/game.rb

```diff
@@ -14,6 +14,30 @@ def populate!
     populate_right_white_half!
   end
 
+  def white_check?
+    white_king = pieces.find_by(type: 'King', color: 'white')
+    pieces.each do |piece|
+      if piece.enemy?(white_king) &&
+         piece.valid_move?(white_king.x_position,
```

Comment:
Edge case that would generate this would be two head-to-head kings. That's kind of a stupid situation, but so are users. Please also add a test for what happens if a king tries to move into a check position with a king